src/bs_size.c

```diff
@@ -663,7 +676,7 @@ char* bs_size_convert_to (const BSSize size, BSUnit unit, BSError **error) {
     }
 
     if (!found_match) {
-        set_error (error, BS_ERROR_INVALID_SPEC, "Invalid unit spec given");
+        set_error (error, BS_ERROR_INVALID_SPEC, strdup ("Invalid unit spec given"));
```

Comment:
Hmm, this should have been causing segfaults, right? How come it was not discovered?

Reply:
Probably because no one is using it directly from C. From python we have a wrapper `convert_to` method that takes string and makes sure you can't pass an invalid value to the C function. But if you try a little harder you can get it to crash even from Python:

```
>>> s._c_size.convert_to(10)      
free(): invalid pointer
Aborted (core dumped)
```

We probably should write some tests in C :-)

Reply:
> And it would be of course harder to pass an invalid unit specification from C, because it is an enum.

There's no such thing as `enum`, it's just `int` anyway.